cmd/csi-attacher/main.go

```diff
@@ -138,14 +138,15 @@ func main() {
 
 	// Prepare http endpoint for metrics + leader election healthz
 	mux := http.NewServeMux()
-	cmm := metrics.NewCSIMetricsManagerForSidecar(csiAttacher)
-	cmm.RegisterToServer(mux, *metricsPath)
-	go func() {
-		err := http.ListenAndServe(*metricsAddress, mux)
-		if err != nil {
-			klog.Fatalf("Failed to start prometheus metrics endpoint on specified address (%q) and path (%q): %s", *metricsAddress, *metricsPath, err)
-		}
-	}()
+	if *metricsAddress != "" {
+		metricsManager.RegisterToServer(mux, *metricsPath)
+		go func() {
```

Comment:
You also need to restore the `metricsManager.SetDriverName(csiAttacher)` line above.